### PR TITLE
update API Reference -> References

### PR DESCRIPTION
**Pull request by cosmicBboy**
Update the API Reference section to contain both API and Component reference sections

Signed-off-by: cosmicBboy <niels.bantilan@gmail.com>

### CODE REVIEW

**Comment by cosmicBboy** [comment]
@kumare3 let me know if organizing the references in this way makes sense

File: rsts/reference/index.rst [hunk 1]
```diff
@@ -40,24 +44,29 @@ API Reference
...
    The Java/Scala SDK for Flyte.
-   
+
    ---
 
-   .. link-button:: https://pkg.go.dev/mod/github.com/flyteorg/flytepropeller
+   .. link-button:: https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/service/service.html
       :type: url
-      :text: FlytePropeller
+      :text: FlyteAdmin
```
Comment (kumare3) [inline rsts/reference/index.rst:52]:
```suggestion
      :text: Flyte Backend REST / gRPC API specification
```

File: rsts/reference/index.rst [hunk 2]
```diff
@@ -40,24 +44,29 @@ API Reference
...
       :classes: btn-block stretched-link
    ^^^^^^^^^^^^
-   The K8s-native operator that executes Flyte workflows.
+   The service responsible for managing Flyte entities and administering workflow executions.
 
-   ---
 
-   .. link-button:: https://pkg.go.dev/mod/github.com/flyteorg/flyteadmin
+Component Reference
```
Comment (kumare3) [inline rsts/reference/index.rst:58]: we should keep Flyteadmin in the component reference

**Comment by kumare3** [comment]
couple comments

**Review by kumare3 — APPROVED** [review]

**Pull request merged**